### DISCUSSION TITLE
Proxy based on user agent headers

### DISCUSSION THREAD

**Discussion by shelbyt** [chimurai/http-proxy-middleware#620] [Q&A] [answered]
Can I route based on the user-agent header? For example, if a header comes in with `googlebot` I want to route the bot to a different url. I only see Context and Path matching in the docs.

**Comment by chimurai** [comment, marked as answer]
You can have a look at the `router` option:

https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/router.md#custom-router-function

```ts
createProxyMiddleware({
  target: 'http://localhost:8000',
  router: function (req) {
    // if googlebot (implement googlebot detection using the req object)
    return 'http://www.example.org'; // different target
  },
})
```

**Reply by shelbyt** [reply]
Thank  you. I'll try that. 

What's the difference between that approach you've shown and this one in the docs using custom matching?:
```
const filter = function (pathname, req) {
  return pathname.match('^/api') && req.method === 'GET';
};

const apiProxy = createProxyMiddleware(filter, {
  target: 'http://www.example.org',
});
```

**Reply by chimurai** [reply]
Those are really two different properties.

`filter` will call express' `next` function to skip further processing.
`router` allows you to assign a different `target`, it's will proxy the call eventually
